### PR TITLE
FOGL-6760: package management fixes for apt and yum

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-6760: initial implementation

### CODE REVIEW

**Review by MarkRiddoch — DISMISSED** [review]

File: python/fledge/services/core/api/update.py [hunk 1]
```diff
@@ -99,4 +100,10 @@ async def update_package(request):
         msg = str(ex)
         raise web.HTTPInternalServerError(reason=msg, body=json.dumps({"message": msg}))
     else:
+        # Save current logged user token
+        token = request.headers.get('authorization', None)
+        if token is not None:
+            with open(os.path.expanduser('~') + '/.fledge_token', 'w') as f:
+                f.write(token)
```
Comment (praveen-garg) [inline python/fledge/services/core/api/update.py:107]: Please move this before server.Server.scheduler.queue_task

**Comment by praveen-garg** [comment]
Also I'm seeing 

INFO: tasks/update: fledge, fledge-gui package updated successfully!
ERROR: tasks/update: Failed on POST Audit trail entry. Found 000 http status code

INFO log stuff will be avoided with upgradable list check.

**Comment by pintomax** [comment]
> Also I'm seeing
> 
> INFO: tasks/update: fledge, fledge-gui package updated successfully! 

ERROR: tasks/update: Failed on POST Audit trail entry. Found 000 http status code

@praveen-garg in bin/update_task.apt routine audit_trail_entry()

there is a POST call without token.

token was removed and after restart it will be no longer valid.

This seems a new issue to solve.

> 
> INFO log stuff will be avoided with upgradable list check.

**Review by MarkRiddoch — DISMISSED** [review]

**Comment by pintomax** [comment]
@praveen-garg please have a look on this as well.

Tested: yum on Centos 7.9
Apt on Ubuntu 18

**Review by MarkRiddoch — APPROVED** [review]

File: scripts/extras/update_task.yum [hunk 2]
```diff
@@ -178,23 +191,43 @@ update_task() {
     fi
 }
 
+# Upgrade check
+upgrade_check() {
+    # System update request
+    run_update
+
+    UPGRADE_CHECK="sudo yum list updates"
```
Comment (praveen-garg) [inline scripts/extras/update_task.yum:199]: sudo is not required. Moreover snap and yum for this release are not supported

File: scripts/extras/update_task.apt [hunk 3]
```diff
@@ -159,32 +178,51 @@ update_task() {
...
+        write_log "" "$0" "debug" "$msg" "all" "pretty"
     fi
 }
 
+# Upgrade check
+upgrade_check() {
+    # System update request
+    run_update
+
+    UPGRADE_CHECK="sudo apt list --upgradable"
```
Comment (praveen-garg) [inline scripts/extras/update_task.apt:194]: sudo not required, but no harm to keep.

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**